sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -305,7 +305,8 @@ class ResolveSessionCatalog(
     // session catalog and the table provider is not v2.
     case c @ ReplaceTableStatement(
          SessionCatalogAndTable(catalog, tbl), _, _, _, _, _, _, _, _, _) =>
-      if (!isV2Provider(c.provider)) {
+      val provider = c.provider.getOrElse(conf.defaultDataSourceName)
```

Comment:
We shouldn't do this change. Since the default provider of a catalog may be different

Reply:
but this is the session catalog? Shouldn't it always use `defaultDataSourceName` to be coherent with sql?

Reply:
I think that implementation needs to live within `V2SessionCatalog`, not here

Reply:
The create table path is already doing this. Shall I change that too then? https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala#L281

Reply:
hmm... 🤔 @rdblue thoughts?

Reply:
I think this should not use Spark's default. The default provider should be determined by the catalog, not by Spark. For example, if my default provider is `parquet` then how should a JDBC catalog interpret this?

Reply:
Sounds reasonable, though I think the AST builder currently always requires a provider in the sql query? I can make the change for create and replace here to pass None in the properties if not specified. @cloud-fan thoughts?

Reply:
It's better to put the default provider in `V2SessionCatalog`, but the problem here is: this rule need to know the provider and see if it's v2, and then pick v1 or v2 command.

Reply:
Maybe `CatalogPlugin` should expose an interface for the default provider? It's not so clear to me what's the relationship between provider and catalog. Should every catalog have a provider?

Reply:
A default provider doesn't make sense for CatalogPlugin, but it can make sense for a `TableCatalog`. V2SessionCatalog then can use the `conf.defaultDataSourceName`. WDYT?

Reply:
The provider doesn't determine whether the command should be run using v1 or v2, the table and catalog do. The provider for the session catalog matters, but not for any other catalog so I don't think changing either the `TableCatalog` or `CatalogPlugin` API is a good idea.

Since this is limited to the v2 session catalog, let's come up with a way to delegate to the v2 session catalog for this. That, or set a default.

Reply:
Can we just say that the existing `defaultDataSourceName` is the default provider for v2 session catalog? Or shall I add another conf for that or add another interface for session catalog?

Reply:
Now that I understand that this problem is scoped to just how we handle the session catalog, I think it's fine to use the v1 default provider here. That's what the v1 and v2 session catalogs would use anyway.

Reply:
yup. @brkyvz @cloud-fan Good to go?

Reply:
yea good to go!